op-supervisor/supervisor/backend/db/db.go

```diff
@@ -162,61 +192,89 @@ func (db *DB) ClosestBlockInfo(blockNum uint64) (uint64, TruncatedHash, error) {
 
 // Contains return true iff the specified logHash is recorded in the specified blockNum and logIdx.
 // logIdx is the index of the log in the array of all logs the block.
-func (db *DB) Contains(blockNum uint64, logIdx uint32, logHash TruncatedHash) (bool, error) {
+func (db *DB) Contains(blockNum uint64, logIdx uint32, logHash TruncatedHash) (bool, ExecutingMessage, error) {
```

Comment:
The query arguments here verify there is an initiating message.

When an executing-link is found, it means that the initiating message was a log-event by the crossL2Inbox that happened to execute something from elsewhere, which has a different logHash (e.g. identifier of executing message points to different chain ID than where it hit the inbox).

So while the return-argument is correctly named `ExecutingMessage`, I think it's quite confusing to return it at all.
`Contains` is only really meant to verify if an initiating message exists. Not to also load the executing message.

I think we should have two methods:
- `func Contains(blockNum, logIdx, logHash) (found bool, err error)`
- `func Executes(blockNum, logIdx) (ExecutingMessage, error)` (return zeroed `ExecutingMessage` if not executing anything)

- `Contains` is used to check if an initiating message exists
- `Executes` is used to check if an executing message exists, which should then be checked to have an initiating counterpart for interop cross-unsafe validation.

And can you include some description like the above in the doc comments of the two functions?

Reply:
I think `Executes` still needs to indicate whether it found the resulting message or not. Otherwise if a log isn't indexed yet, it will be incorrectly interpreted as not having an `ExecutingMessage` which may not be accurate.  Will get it to return `ErrNotFound` if the specified log isn't in the database.

Reply:
Split out into two methods.